fujiji-client/src/pages/search/index.jsx

```diff
@@ -0,0 +1,101 @@
+import {
+  Box, Center, SimpleGrid, Text, Spinner,
+} from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+import { AdListingCard, withSession } from '../../components';
+import { getAllListings } from '../../server/api';
+
+function SearchPage() {
+  const router = useRouter();
+  const [listings, setListings] = useState([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  // for this page we are using client-side rendering
+  // so we can show loading indicator when fetching huge list of listings
+  useEffect(() => {
+    async function fetchData() {
+      const res = await getAllListings();
+      if (res.error) {
+        setErrorMessage(res.error);
+      } else {
+        setListings(res);
+      }
+    }
+
+    fetchData();
+  }, []);
+
+  let renderListings;
+
+  if (listings.length > 0) {
+    // There are listings that exist for this user
+    const listOfListings = listings.map((listing) => {
+      const onSubmit = () => {
+        router.push(`/listing/${listing.listingID}`);
+      };
+      return (
+        <Box id={`listing=${listing.listingID}`} key={listing.listingID}>
+          <AdListingCard {...listing} onClick={onSubmit} />
+        </Box>
+      );
+    });
+
+    renderListings = (
+      <SimpleGrid
+        id="userListingsContent"
+        columns={[1, 2, null, null, 3]}
+        spacing="10"
+      >
+        {listOfListings}
+      </SimpleGrid>
+    );
+  } else if (errorMessage) {
+    // User does not have any listings yet
+    renderListings = <Text>{errorMessage}</Text>;
+  } else {
+    // Loading indicator while fetching
+    renderListings = (
+      <Center>
+        <Spinner
+          thickness="4px"
+          speed="0.65s"
+          emptyColor="gray.200"
+          color="blue.500"
+          size="lg"
+          mr="4"
+        />
+        <Text id="testId">We are getting your listings...</Text>
+      </Center>
+    );
+  }
+
+  return (
+
+    <Center px="1" py="6" flexDir="column">
+      <Text fontSize="36px" fontWeight="bold" mb="6">
+        Search
+      </Text>
+      {renderListings}
+    </Center>
+  );
+}
+
+export default withSession(SearchPage);
```

Comment:
Can we remove `withSession()` here? Since a user without account can also visit the search page.

So we can just:
```
export default SearchPage
```

Reply:
This reminds me I put `withSession()` in `/listing/[id]` page too. And it's not required lol: https://github.com/devin-efendy/fujiji/blob/b83774033234189149876d691637c0a0449e58ac/fujiji-client/src/pages/listing/%5Bid%5D.jsx#L18